include/aspect/particle/property/viscoplastic_strain_invariants.h

```diff
@@ -91,6 +91,12 @@ namespace aspect
         private:
           unsigned int n_components;
 
+          /**
+           * An object that is used to compute the particle property. Since the
+           * object is expensive to create and is needed often it is kept as a
+           * member variable.
+           */
+          mutable MaterialModel::MaterialModelInputs<dim> material_inputs;
```

Comment:
Is it guaranteed that we are not doing any updates from multiple threads? I think this is not called from assembly right now, but somebody might want to do particle updates from multiple threads. Not sure how important that is right now, but it is worth thinking about.

Reply:
@gassmoeller - could you give a quick summary of why the object needs to be defined as `mutable`?